module/questions/helper.js

```diff
@@ -842,4 +842,113 @@ module.exports = class QuestionsHelper {
     }) 
   }
 
+
+  /**
+   * Create copy of questions.
+   * @method
+   * @name createCopyOfQuestions
+   * @param {Array} questionIds - Array of question Id's         
+   * @returns {Object}  old and new Mapped question ids .  
+  */
+
+  static createCopyOfQuestions(questionIds = []) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        if (!questionIds.length) {
+          throw new Error(messageConstants.apiResponses.QUESTION_ID_REQUIRED)
+        }
+
+        let questionIdMap = {};
+
+        let questionDocuments = await this.questionDocument
+          (
+            { _id: { $in: questionIds } }
+          )
+
+        if (!questionDocuments.length) {
+          throw new Error(messageConstants.apiResponses.QUESTION_NOT_FOUND)
+        }
+
+        await Promise.all(questionDocuments.map(async question => {
```

Comment:
@deepakr123 - We should handle cases of one question in multiple criteria, in which case duplicate should only be once. Also what if the instance child is already duplicated in some previous criteria?

Reply:
resolved